unit/eq_test.c

```diff
@@ -532,6 +533,45 @@ eq_wait_fd_sread()
 	return TEST_RET_VAL(ret, testret);
 }
 
+static int
+eq_readerr_ret_eagain() {
+
+	int testret;
+	int ret;
+	struct fi_eq_err_entry err_entry;
+
+	testret = FAIL;
+
+	memset(&err_entry, 0, sizeof(err_entry));
+
+	/* create EQ  */
+	ret = create_eq(32, FI_WRITE, FI_WAIT_UNSPEC);
+	if (ret) {
+		sprintf(err_buf, "fi_eq_open ret=%d, %s", ret, fi_strerror(-ret));
+		goto fail;
+	}
+
+	ret = insert_events(5);
+	if (ret)
+		goto fail;
+
+	ret = read_events(5, 0);
+	if (ret)
+		goto fail;
+
+	/* Error handling  */
+	ret = fi_eq_readerr(eq, &err_entry, 0);
+	if (ret == -FI_EAGAIN) {
+		sprintf(err_buf, ", fi_eq_readerr returned: %d: %s, ", ret,
+				fi_eq_strerror(eq, err_entry.prov_errno, err_entry.err_data, NULL, 0));
```

Comment:
This is the expected success path, so the print isn't needed here.  It would be better as part of an else condition as a failure.

Reply:
Done.